src/base/moduleresolver.lua

```diff
@@ -0,0 +1,90 @@
+--
+-- moduleresolver.lua
+-- 
+-- Provides a strategy for module resolution to modules that are placed within the default module's directory. 
+-- 
+-- Copyright (c) 2002-2014 Jason Perkins and the Premake project
+--
+
+    local p = premake
+
+    p.moduleresolver = {}
+
+    local moduleresolver = p.moduleresolver
+
+---
+-- Check whether the module is either a path to a module or a module's name.
+--
+
+    function moduleresolver.isModulePath(module)
+        return string.contains(module, "/") or string.contains(module, "\\")
+    end
+    
+--
+-- Get the name of the default module's directory.
+--
+
+    function moduleresolver.getDefaultModuleDirectory()
+        return ".premake"
+    end
+    
+---
+-- Get the path to the default module's directory.
+--
+
+    function moduleresolver.getDefaultModulePath(modulePath)
+        return modulePath and path.join(moduleresolver.getDefaultModuleDirectory(), path.appendExtension(modulePath, ".lua")) or modulePath
```

Comment:
I'll add it, would it be okay if I'll add it in addition to .premake/module.lua?